### PR TITLE
Adding ballooning support

### PR DESCRIPTION
**Pull request by janosdebugs**
## Please describe the change you are making

Adding support for memory ballooning. It also changes the return of the `MemoryPolicy()` function as I discovered that it is always present.

## Are you the owner of the code you are sending in, or do you have permission of the owner?

Yes

## The code will be published under the BSD 3 clause license. Have you read and understood this license?

Yes

### CODE REVIEW

**Review by eslutsky — COMMENTED** [review]
can we implement any tests for this ?

**Comment by janosdebugs** [comment]
> can we implement any tests for this ?

done

File: vm.go [hunk 1]
```diff
@@ -215,7 +215,7 @@ type VMData interface {
 	Memory() int64
 	// MemoryPolicy returns the memory policy set on the VM, if any. The second parameter returned is true if the
```
Comment (engelmi) [inline vm.go:216]: This inline-doc needs to be adjusted, I think.

File: vm_test.go [hunk 2]
```diff
@@ -527,6 +537,58 @@ func TestMaxMemory(t *testing.T) {
...
+			vm := assertCanCreateVM(
+				t,
+				helper,
+				helper.GenerateTestResourceName(t),
+				params,
+			)
+			memoryPolicy := vm.MemoryPolicy()
+			ballooning := memoryPolicy.Ballooning()
+			if ballooning != testCase.expected {
+				t.Fatalf("Incorrect ballooning value")
```
Comment (engelmi) [inline vm_test.go:585]: Minor: If we make this `Fatalf` more expressive, it'll be easier to figure out what is wrong. E.g.
```go
t.Fatalf("Expected balooning value to be %t, but got %t", testCase.expected, balooning)
```

**Review by engelmi — APPROVED** [review]

**Pull request merged**